### PR TITLE
Bump typescript and @salesforce/core versions

### PR DESCRIPTION
**Pull request by allileong**
* Update typescript to 3.1.6 to match @salesforce/core
* Update @salesforce/core to 1.0.1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "devDependencies": {
-    "@salesforce/dev-config": "1.1.0",
+    "@salesforce/dev-config": "1.4.1",
```
Comment (allileong) [inline package.json]: Updating @salesforce/dev-config, since one of its dependencies has a peer dependency on typescript

File: package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "shx": "0.2.2",
     "snyk": "^1.82.1",
     "tslint": "5.5.0",
-    "typescript": "2.6.2",
+    "typescript": "3.1.6",
```
Comment (allileong) [inline package.json]: This is the same version of typescript used by @salesforce/core 1.0.1

File: packages/salesforcedx-apex-debugger/tsconfig.json [hunk 3]
```diff
@@ -2,22 +2,16 @@
...
     "sourceMap": true,
     "declaration": true,
     "moduleResolution": "node",
     "noImplicitAny": true,
     "rootDir": ".",
     "outDir": "out",
     "preserveConstEnums": true,
-    "strict": true
+    "strict": true,
+    "strictPropertyInitialization": false
```
Comment (allileong) [inline packages/salesforcedx-apex-debugger/tsconfig.json]: I opened work item W-5676805 to refactor our code to conform to the "strictPropertyInitialization" rule.

File: tslint.json [hunk 4]
```diff
@@ -6,6 +6,7 @@
     "quotemark": [true, "single", "avoid-escape"],
     "member-ordering": false,
     "no-string-based-set-timeout": false,
-    "semicolon": [true, "always", "ignore-bound-class-methods"]
+    "semicolon": [true, "always", "ignore-bound-class-methods"],
+    "no-floating-promises": false
```
Comment (allileong) [inline tslint.json]: The new version of @salesforce/dev-config has `"no-floating-promises": true`. I opened the work item W-5676815 to refactor our code to conform to this rule.

**Comment by allileong** [comment]
@lcampos @vazexqi , may I be granted access to see the security/snyk failure details above? This is what I see when I try to click on the details link: 
screen shot 2018-12-05 at 5 34 21 pm

**Comment by vazexqi** [comment]
@allileong - I sent out an invite for synk to your `@salesforce.com` e-mail.

Thanks for taking the initiative to bump us up to TypeScript 3.x. There are lots of good features here that we can take advantage of.

File: packages/salesforcedx-vscode-visualforce/package.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
   "dependencies": {
     "@salesforce/salesforcedx-visualforce-language-server": "44.10.0",
     "@salesforce/salesforcedx-visualforce-markup-language-server": "44.10.0",
-    "typescript": "2.6.2",
+    "typescript": "3.1.6",
```
Comment (lcampos) [inline packages/salesforcedx-vscode-visualforce/package.json]: We should move this to `devDependencies`.

File: packages/salesforcedx-apex-debugger/tsconfig.json (hunk 3 above)
Comment (dehru) [reply packages/salesforcedx-apex-debugger/tsconfig.json]: @allileong - I'm going to work on W-5676805 in GUS.  Do you need me to open up an issue for that on github, or is it sufficient for me to reference W-5676805 in my PR without a matching Github Issue?

Comment (allileong) [reply packages/salesforcedx-apex-debugger/tsconfig.json]: @dehru - There's no need to create a separate github issue. Referencing W-5676805 in your PR will be sufficient. Thanks for working on this item!

**Pull request merged**